internal/handlers/handlers.go

```diff
@@ -95,21 +188,245 @@ func (h *handler) GetMetric() http.HandlerFunc {
 			return
 		}
 
-		res.Write([]byte(metric))
+		size, err := res.Write([]byte(metric))
+		if err != nil {
+			h.Logger.Error("Error res.Write(metricsJSON)",
+				zap.String("about func", "GetMetric"),
+				zap.String("about ERR", err.Error()),
+			)
+		}
 
 		res.WriteHeader(http.StatusOK)
+
+		h.Logger.Info("sending HTTP response UpdatedMetric",
+			zap.String("size", strconv.Itoa(size)),
+			zap.String("status", strconv.Itoa(http.StatusOK)),
+		)
+	}
+}
+
+func (h *handler) UpdatedJSONMetric() http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+
+		h.Logger.Debug("decoding request")
+		var metrics models.Metrics
+		dec := json.NewDecoder(req.Body)
+		if err := dec.Decode(&metrics); err != nil {
+			h.Logger.Debug("cannot decode request JSON body", zap.Error(err))
+			res.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		deltaDefault := int64(0)
```

Comment:
лучше тогда так делать
```suggestion
		var deltaDefault int64
```